File: EmbASP_Desktop_BlocksWorld_Python/src/BlocksWorld.py
from platforms.desktop.desktop_handler import DesktopHandler
from specializations.solver_planning_domains.desktop.spd_desktop_service import SPDDesktopService
from languages.pddl.pddl_input_program import PDDLInputProgram
from languages.pddl.pddl_program_type import PDDLProgramType
from languages.pddl.pddl_mapper import PDDLMapper
from PickUp import PickUp
from PutDown import PutDown
from Stack import Stack
from Unstack import Unstack

handler  = DesktopHandler(SPDDesktopService())

input_domain = PDDLInputProgram(PDDLProgramType.DOMAIN)
input_domain.add_files_path("../domain.pddl")

input_problem= PDDLInputProgram(PDDLProgramType.PROBLEM)
input_problem.add_files_path("../p01.pddl")

handler.add_program(input_domain)
handler.add_program(input_problem)

PDDLMapper.get_instance().register_class(PickUp)
PDDLMapper.get_instance().register_class(PutDown)
PDDLMapper.get_instance().register_class(Stack)
PDDLMapper.get_instance().register_class(Unstack)

output = handler.start_sync()

for obj in output.get_actions_objects():
    if isinstance(obj, PickUp) | isinstance(obj, PutDown) | isinstance(obj, Stack) | isinstance(obj, Unstack) :
        print(obj)

File: EmbASP_Desktop_BlocksWorld_Java/src/it/mat/unical/embasp/Blocksworld.java
package it.mat.unical.embasp;

import it.unical.mat.embasp.base.Handler;
import it.unical.mat.embasp.base.InputProgram;
import it.unical.mat.embasp.base.Output;
import it.unical.mat.embasp.languages.pddl.PDDLInputProgram;
import it.unical.mat.embasp.languages.pddl.PDDLMapper;
import it.unical.mat.embasp.languages.pddl.PDDLProgramType;
import it.unical.mat.embasp.languages.pddl.Plan;
import it.unical.mat.embasp.platforms.desktop.DesktopHandler;
import it.unical.mat.embasp.specializations.solver_planning_domains.desktop.SPDDesktopService;

public class Blocksworld {
	
	private static String domainFileName = "domain.pddl";
	private static String problemFileName = "p01.pddl";;

	public static void main(String[] args) {
		Handler handler = new DesktopHandler(new SPDDesktopService());
		
		final InputProgram inputProgramDomain = new PDDLInputProgram(PDDLProgramType.DOMAIN);
		inputProgramDomain.addFilesPath(domainFileName);

		final InputProgram inputProgramProblem = new PDDLInputProgram(PDDLProgramType.PROBLEM);
		inputProgramProblem.addFilesPath(problemFileName);

		handler.addProgram(inputProgramDomain);
		handler.addProgram(inputProgramProblem);

		try {

			PDDLMapper.getInstance().registerClass(PickUp.class);
			PDDLMapper.getInstance().registerClass(PutDown.class);
			PDDLMapper.getInstance().registerClass(Stack.class);
			PDDLMapper.getInstance().registerClass(Unstack.class);

			Output output = handler.startSync();
			
			if (!(output instanceof Plan))
				return; 
			
			Plan plan = (Plan) output;
			
			for (final Object obj : plan.getActionsObjects())
				if (obj instanceof PickUp || obj instanceof Stack
						 || obj instanceof Unstack || obj instanceof PutDown) 
					System.out.println(obj.toString());
		
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

